src/middlewared/middlewared/plugins/kubernetes_linux/migrations/0001_docker_overalyfs.py

```diff
@@ -0,0 +1,27 @@
+import os
+
+
+def migrate(middleware):
+    # We would be doing the following here:
+    # 1) Remove old docker dataset
+    # 2) Create new docker dataset
+    # 3) Start docker
+    # 4) Load bundled docker images
+    # 5) Call it a day
+    k8s_config = middleware.call_sync('kubernetes.config')
+    docker_ds = os.path.join(k8s_config['dataset'], 'docker')
+    if middleware.call_sync(
+        'zfs.dataset.get_instance', docker_ds, {
```

Comment:
This will error out as `zfs.dataset.get_instance` will raise an error if the dataset does not exist